### PR TITLE
Refactor Jenkins to use mock file on OSX and lock file only for mutations

### LINKED ISSUE #996: Investigate why safe_app tests can run slow with on-disk mock vault storage
Sometimes the safe_app tests can run so slow they time out during the entire test suite run.

There are several possible causes for that, but it's worth starting with one of these:
1. The Vault locking logic is inefficient.
2. The storage itself gets overflowed and causes the slowdowns.

### PR DESCRIPTION
**Pull request by lionel1704**
Hey guys, 
I wanted to close and re-open #999 to re-trigger the build since GH is having issues, but I'm unable to re-open it now. So here's a new one.

closes #996

### CODE REVIEW

File: safe_core/src/client/mock/connection_manager.rs [hunk 1]
```diff
@@ -63,7 +63,7 @@ impl ConnectionManager {
     /// Send `message` via the `ConnectionGroup` specified by our given `pub_id`.
     pub fn send(&mut self, pub_id: &PublicId, msg: &Message) -> Box<CoreFuture<Response>> {
         let msg: Message = {
-            let mut vault = vault::lock(&self.vault, true);
+            let mut vault = vault::lock(&self.vault, is_read(msg));
```
Comment (m-cat) [inline safe_core/src/client/mock/connection_manager.rs]: Correct me if I'm wrong but shouldn't this be `!is_read` (as this is the `writing` parameter)?

File: safe_core/src/client/mock/connection_manager.rs [hunk 2]
```diff
@@ -140,6 +140,61 @@ impl ConnectionManager {
     }
 }
 
+fn is_read(msg: &Message) -> bool {
```
Comment (m-cat) [inline safe_core/src/client/mock/connection_manager.rs]: Possible to reuse `request_is_get()` instead of introducing a new, similar function?

**Comment by lionel1704** [comment]
You're right @m-cat !
I'll refactor this :)

**Comment by m-cat** [comment]
Superseded by https://github.com/maidsafe/safe_client_libs/pull/1006.

**Pull request closed without merging**